cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -309,9 +309,25 @@ func (p *AzdoScmProvider) configureGitRemote(ctx context.Context, repoPath strin
 			return "", err
 		}
 	}
+
+	branch, err := p.getCurrentGitBranch(ctx, repoPath)
+	if err != nil {
+		return "", err
+	}
+	azdo.DefaultBranch = branch
+
 	return remoteUrl, nil
 }
 
+func (p *AzdoScmProvider) getCurrentGitBranch(ctx context.Context, repoPath string) (string, error) {
+	gitCli := git.NewGitCli(ctx)
+	branch, err := gitCli.GetCurrentBranch(ctx, repoPath)
+	if err != nil {
+		return "", err
+	}
+	return branch, nil
+}
+
```

Comment:
Sure , with the extra method it makes it easier to change or debug the underlying implementation. It's a simple shim and I tend to avoid heavy method chaining and follow the law of Demeter.

https://en.m.wikipedia.org/wiki/Law_of_Demeter

Reply:
Generally, not my style :)

And I think the `law of demeter` tries to `loosen coupling`, and in this case, you are adding the `wrapper` method as a `func (p *AzdoScmProvider)`, so it is still coupled the same way, as it is the same `AzdoScmProvider`

We are already handling the coupling by doing dependency injection.  The `gitCli` interface is injected using the context and the `gitCli` is already the `wrapper` (or facade) that hides the logic behind.